platforms/core-configuration/classloading/build.gradle.kts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+plugins {
+    id("gradlebuild.distribution.api-java")
+}
+
+description = "Class loader scopes, registries and instrumentation"
+
+// TODO: ensure code is compliant with requirements for workers
```

Comment:
We should avoid dependencies on :core-api as well as dependencies on :core, as :core-api classes are implemented in :core and require it at runtime

Reply:
The dependency on `:core-api` was not introduced, but rather "exposed" by moving the classes. I agree we should avoid it, and eventually not have `:core-api` module at all as it should be split into meaningful pieces. For the moment, I don't think it's necessary to resolve it.

I tried removing the dependency, and compilation of the `:classloading` passes successfully, as well as some integration tests. It seems the only reason why this dependency is present is because it's a transitive one and the Dependency Analysis requires it listed. We may not be able to avoid mentioning it until it stops being a transitive dependency, which can be much harder to achieve.

```
These transitive dependencies should be declared directly:
  api(project(":core-api"))
```